test/e2e/framework/nodes_util.go

```diff
@@ -394,3 +396,23 @@ func (k *NodeKiller) kill(nodes []v1.Node) {
 	}
 	wg.Wait()
 }
+
+func DeleteNodeOnCloudProvider(node *v1.Node) error {
+	return TestContext.CloudConfig.Provider.DeleteNode(node)
+}
+
+func WaitForNodeToBeDeleted(c clientset.Interface, node *v1.Node, timeout time.Duration) (bool, error) {
+	var err error
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+		_, err = c.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+		if apierrs.IsNotFound(err) {
+			return true, nil
+		}
+	}
+
+	if err == nil {
+		return false, nil
+	}
+
+	return false, fmt.Errorf("failed to check if node %q was deleted: %q", node.Name, err)
```

Comment:
Originally didn't do this because it allowed the request to be retried, but I'm noticing there's a `testutils.IsRetryableAPIError` I can use to check if the error is retryable so will use that :)